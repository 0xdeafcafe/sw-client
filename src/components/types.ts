export type DisplayProps = {
  currentRoot: string;
  item: FilmItemProps &
    PersonItemProps &
    PlanetItemProps &
    SpeciesItemProps &
    StarshipItemProps &
    VehicleItemProps;
};

export type FilmProps = {
  item: FilmItemProps;
  isItemFavourite: boolean;
  setIsItemFavourite: (boolean: boolean) => void;
};

export type FilmItemProps = {
  characters: string[];
  created: string;
  director: string;
  edited: string;
  episode_id: number;
  opening_crawl: string;
  planets: string[];
  producer: string;
  release_date: string;
  species: string[];
  starships: string[];
  title: string;
  url: string;
  vehicles: string[];
};

export type PersonProps = {
  item: PersonItemProps;
  isItemFavourite: boolean;
  setIsItemFavourite: (boolean: boolean) => void;
};

export type PersonItemProps = {
  created: string;
  edited: string;
  birth_year: string;
  eye_color: string;
  films: string[];
  gender: string;
  hair_color: string;
  height: string;
  homeworld: string;
  name: string;
  skin_color: string;
  species: string[];
  starships: string[];
  url: string;
  vehicles: string[];
  mass: string;
};

export type PlanetProps = {
  item: PlanetItemProps;
  isItemFavourite: boolean;
  setIsItemFavourite: (boolean: boolean) => void;
};

export type PlanetItemProps = {
  name: string;
  diameter: string;
  orbital_period: string;
  population: string;
  rotation_period: string;
  surface_water: string;
  gravity: string;
  terrain: string;
  climate: string;
  films: string[];
  residents: string[];
  url: string;
};

export type SpeciesProps = {
  item: SpeciesItemProps;
  isItemFavourite: boolean;
  setIsItemFavourite: (boolean: boolean) => void;
};

export type SpeciesItemProps = {
  name: string;
  average_height: string;
  average_lifespan: string;
  classification: string;
  designation: string;
  eye_colors: string;
  hair_colors: string;
  language: string;
  skin_colors: string;
  films: string[];
  people: string[];
  url: string;
};

export type StarhsipProps = {
  item: StarshipItemProps;
  isItemFavourite: boolean;
  setIsItemFavourite: (boolean: boolean) => void;
};

export type StarshipItemProps = {
  name: string;
  model: string;
  manufacturer: string;
  cargo_capacity: string;
  hyperdrive_rating: string;
  consumables: string;
  cost_in_credits: string;
  crew: string;
  length: string;
  max_atmosphering_speed: string;
  passengers: string;
  starship_class: string;
  films: string[];
  pilots: string[];
  url: string;
};

export type VehicleProps = {
  item: VehicleItemProps;
  isItemFavourite: boolean;
  setIsItemFavourite: (boolean: boolean) => void;
};

export type VehicleItemProps = {
  name: string;
  model: string;
  manufacturer: string;
  cargo_capacity: string;
  consumables: string;
  cost_in_credits: string;
  crew: string;
  length: string;
  max_atmosphering_speed: string;
  passengers: string;
  vehicle_class: string;
  films: string[];
  pilots: string[];
  url: string;
};

export type AccordionProps = {
  item: FilmItemProps &
    PersonItemProps &
    PlanetItemProps &
    SpeciesItemProps &
    StarshipItemProps &
    VehicleItemProps;
  currentRoot: string;
};

export type ModalProps = {
  displayName: String;
  result: () => React.ReactNode;
  item: string[];
};

export type State = {
  roots: {
    isLoading: boolean;
    isBranchedRootLoading: boolean;
    payload: object;
    error: string[];
    currentRoot: StateCurrentRoot;
    branchedRoot: FilmItemProps[] &
      PersonItemProps[] &
      PlanetItemProps[] &
      SpeciesItemProps[] &
      StarshipItemProps[] &
      VehicleItemProps[];
  };
};

type StateCurrentRoot = {
  count: number;
  next: string;
  previous: string | null;
  results: FilmItemProps[] &
    PersonItemProps[] &
    PlanetItemProps[] &
    SpeciesItemProps[] &
    StarshipItemProps[] &
    VehicleItemProps[];
  k: string;
};
