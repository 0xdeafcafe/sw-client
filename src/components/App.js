import React from 'react';
import { Container, Navbar, NavbarBrand } from 'reactstrap';

const App = ({ children }) => (
	<div className={'App'}>
		<Navbar
			color={'light'}
			light
		>
			<NavbarBrand href={'/'}>{'Star Wars Client'}</NavbarBrand>
		</Navbar>

		<Container className={'pt-5'}>
			{children}
		</Container>
	</div>
);

export default App;
