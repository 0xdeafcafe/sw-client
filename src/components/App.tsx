
import React, {ReactNode} from 'react';
import { Container, Navbar, NavbarBrand } from 'reactstrap';

interface AppProps {
	children: ReactNode;
  }

const App: React.FC<AppProps> = ({ children }) => (
	<div className={'App'}>
		<Navbar
			color={'light'}
			light
		>
			<NavbarBrand href={'/'}>{'Star Wars Client'}</NavbarBrand>
		</Navbar>

		<Container className={'pt-5'}>
			{children}
		</Container>
	</div>
);

export default App;
