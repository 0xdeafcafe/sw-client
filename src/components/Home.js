import classnames from 'classnames';
import { fetchRoots } from '../actions';
import sentenceCase from 'sentence-case';
import {
	Nav,
	NavItem,
	NavLink,
	TabContent,
	TabPane,
} from 'reactstrap';
import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';

const Home = () => {
	const dispatch = useDispatch();
	const [tab, setTab] = useState(void 0);
	const roots = useSelector(state => state.roots);

	useEffect(() => {
		dispatch(fetchRoots());
	}, [dispatch]);

	const keys = Object.keys(roots.payload || {});

	return (
		<div>
			<h1>{'My little Star Wars app ðŸ‘¾'}</h1>

			{roots.payload && (
				<div className={'mt-3'}>
					<Nav tabs>
						{keys.map(k => (
							<NavItem key={k}>
								<NavLink
									className={classnames({ active: tab === k })}
									onClick={() => setTab(k)}
								>
									{sentenceCase(k)}
								</NavLink>
							</NavItem>
						))}
					</Nav>

					<TabContent activeTab={tab}>
						{keys.map(k => (
							<TabPane
								key={k}
								tabId={k}
							>
								<code>{'todo: load resource table'}</code>
							</TabPane>
						))}
					</TabContent>
				</div>
			)}
		</div>
	);
};

export default Home;
