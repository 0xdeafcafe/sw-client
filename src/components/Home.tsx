import classnames from "classnames";
import sentenceCase from "sentence-case";
import styled from "styled-components";
import { fetchIndividualRoot, fetchRoots } from "../actions/actions";
import { Nav, NavItem, NavLink, TabContent, TabPane, Button } from "reactstrap";
import CircularProgress from "@mui/material/CircularProgress";

import React, { useEffect, useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import { ItemAccordion } from "./Accordion/Accordion";
import { FilmItemProps, PersonItemProps, PlanetItemProps, SpeciesItemProps, StarshipItemProps, VehicleItemProps } from "./types";

const SpinnerContainer = styled.div`
  display: flex;
  justify-content: center;
  padding: 50px;
`;

const StyledButtonContainer = styled.div`
  display: flex;
  justify-content: flex-end;
`

const Home = () => {
  const dispatch = useDispatch();
  const [tab, setTab] = useState<string | undefined>(void 0);

  const roots = useSelector((state: any) => state.roots);
  const isLoading = useSelector((state: any) => state.roots.isLoading);
  const currentRoot = useSelector(
    (state: any) =>
      (state &&
        state.roots &&
        state.roots.currentRoot &&
        state.roots.currentRoot.k) ||
      []
  );

  const currentRootResults = useSelector(
    (state: any) =>
      (state &&
        state.roots &&
        state.roots.currentRoot &&
        state.roots.currentRoot) ||
      []
  );

  useEffect(() => {
    dispatch(fetchRoots());
  }, [dispatch]);

  const keys = Object.keys(roots.payload || {});

  const handleTabClick = (k: string) => {
    dispatch(fetchIndividualRoot(k));
  };
  return (
    <div>
      <h1>{"My little Star Wars app ðŸ‘¾"}</h1>

      {roots.payload && (
        <div className={"mt-3"}>
          <Nav tabs>
            {keys.map((k) => (
              <NavItem key={k}>
                <NavLink
                  className={classnames({ active: tab === k })}
                  onClick={() => {
                    setTab(k);
                    handleTabClick(k);
                  }}
                >
                  {sentenceCase(k)}
                </NavLink>
              </NavItem>
            ))}
          </Nav>

          {isLoading ? (
            <SpinnerContainer>
              <CircularProgress />
            </SpinnerContainer>
          ) : (
            <>
              <TabContent activeTab={tab}>
                {keys.map((k) => (
                  <TabPane key={k} tabId={k}>
                    {currentRootResults && currentRootResults.results && currentRootResults.results.map((item: FilmItemProps & PersonItemProps & PlanetItemProps & SpeciesItemProps & StarshipItemProps & VehicleItemProps) => (
                      <ItemAccordion currentRoot={currentRoot} item={item} />
                    ))}
                  </TabPane>
                ))}
                <StyledButtonContainer>
                  {currentRootResults.previous &&<Button onClick={() => dispatch(fetchIndividualRoot(currentRootResults.previous))} >Previous</Button>}
                  {currentRootResults.next &&<Button onClick={() => dispatch(fetchIndividualRoot(currentRootResults.next))}>Next</Button>}
                </StyledButtonContainer>
              </TabContent>
            </>
          )}
        </div>
      )}
    </div>
  );
};

export default Home;
