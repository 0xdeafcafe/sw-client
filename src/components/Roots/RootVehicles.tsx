import { State, VehicleProps } from "components/types";
import React from "react";
import { useSelector } from "react-redux";
import Modal from "../Modal/Modal";
import FavoriteBorderIcon from "@mui/icons-material/FavoriteBorder";
import FavoriteIcon from "@mui/icons-material/Favorite";
import styled from "styled-components";
import { removeFromLocalStore, SaveDataToLocalStorage } from "utils/utils";

const StyledIconContainer = styled.div`
  margin-left: 3px;
  display: flex;
  justify-content: flex-end;
  cursor: pointer;
`;

export const RootVehicles = ({
  item: {
    name,
    model,
    manufacturer,
    cargo_capacity,
    consumables,
    cost_in_credits,
    crew,
    length,
    max_atmosphering_speed,
    passengers,
    vehicle_class,
    films,
    pilots,
    url,
  },
  isItemFavourite,
  setIsItemFavourite,
}: VehicleProps) => {
  const branchedRoot = useSelector(
    (state: State) =>
      (state &&
        state.roots &&
        state.roots.branchedRoot &&
        state.roots.branchedRoot) ||
      []
  );

  const result = () =>
    branchedRoot.map((item: { title: string }) => <p>{item.title}</p>);

  const residentsResult = () =>
    branchedRoot.map((item: { name: string; gender: string }) => (
      <div>
        <p>{`Resident Name: ${item.name}`}</p>
        <p>{`Gender: ${item.gender}`}</p>
        <hr />
      </div>
    ));

  return (
    <div>
      <StyledIconContainer
        onClick={() => {
          if (isItemFavourite) {
            removeFromLocalStore(url);
            setIsItemFavourite(false);
          } else {
            SaveDataToLocalStorage(url);
            setIsItemFavourite(true);
          }
        }}
      >
        {isItemFavourite ? <FavoriteIcon /> : <FavoriteBorderIcon />}
      </StyledIconContainer>
      <div>
        <p>{`Vehicle Name: ${name}`}</p>
        <p>{`Vehicle Model: ${model}`}</p>
        <p>{`Manufacturer: ${manufacturer}`}</p>
        <p>{`Cargo Capacity: ${cargo_capacity}`}</p>
        <p>{`Consumables: ${consumables}`}</p>
        <p>{`Cost in Credits: ${cost_in_credits}`}</p>
        <p>{`Crew Size: ${crew}`}</p>
        <p>{`Length: ${length}`}</p>
        <p>{`Maximum Atmospheric Speed: ${max_atmosphering_speed}`}</p>
        <p>{`Passengers: ${passengers}`}</p>
        <p>{`Vehicle Class: ${vehicle_class}`}</p>
      </div>

      {films.length !== 0 && (
        <Modal displayName="Films" item={films} result={result} />
      )}

      {pilots.length !== 0 && (
        <Modal displayName="Pilots" item={pilots} result={residentsResult} />
      )}
    </div>
  );
};
