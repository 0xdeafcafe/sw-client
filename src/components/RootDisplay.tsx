import React, {useState, useEffect} from "react";
import { RootPerson } from "./Roots/RootPerson";
import { RootPlanets } from "./Roots/RootPlanets";
import { RootSpecies } from "./Roots/RootSpecies";
import { RootStarships } from "./Roots/RootStarships";
import { RootVehicles } from "./Roots/RootVehicles";
import { RootFilms } from "./Roots/RootFilms";
import { DisplayProps } from "./types";

export const RootDisplay = ({ item, item: {url}, currentRoot }: DisplayProps) => {
  const [isItemFavourite, setIsItemFavourite] = useState(false)

  useEffect(() => {
    const localStorageItems = JSON.parse(localStorage.getItem("session") || '[]')

    if(localStorageItems.includes(url)) {
      setIsItemFavourite(true)
    }
  })

  return(
  <>
    {currentRoot === "people" && <RootPerson setIsItemFavourite={setIsItemFavourite} isItemFavourite={isItemFavourite} item={item} />}
    {currentRoot === "planets" && <RootPlanets setIsItemFavourite={setIsItemFavourite} isItemFavourite={isItemFavourite} item={item} />}
    {currentRoot === "films" && <RootFilms setIsItemFavourite={setIsItemFavourite} isItemFavourite={isItemFavourite} item={item} />}
    {currentRoot === "species" && <RootSpecies setIsItemFavourite={setIsItemFavourite} isItemFavourite={isItemFavourite} item={item} />}
    {currentRoot === "vehicles" && <RootVehicles setIsItemFavourite={setIsItemFavourite} isItemFavourite={isItemFavourite}item={item} />}
    {currentRoot === "starships" && <RootStarships setIsItemFavourite={setIsItemFavourite} isItemFavourite={isItemFavourite} item={item} />}
  </>
);
}