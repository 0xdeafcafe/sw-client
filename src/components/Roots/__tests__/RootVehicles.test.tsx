import { VehicleProps } from "components/types";
import React from "react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import initialState from "reducers/initial-state";

import { configure, mount } from "enzyme";

import Adapter from "enzyme-adapter-react-16";
import { RootVehicles } from "../RootVehicles";

configure({ adapter: new Adapter() });

const mockStore = configureStore();
const store = mockStore(initialState);

const props: VehicleProps = {
  item: {
    name: "",
    model: "",
    manufacturer: "",
    cargo_capacity: "",
    consumables: "",
    cost_in_credits: "",
    crew: "",
    length: "",
    max_atmosphering_speed: "",
    passengers: "",
    vehicle_class: "",
    films: ["test"],
    pilots: ["test"],
    url: "",
  },
  isItemFavourite: false,
  setIsItemFavourite: (boolean: boolean) => null,
};

it("renders correctly when there are no items", () => {
  const wrapper = mount(
    <Provider store={store}>
      <RootVehicles {...props} />
    </Provider>
  );
  expect(wrapper).toMatchSnapshot();
});
