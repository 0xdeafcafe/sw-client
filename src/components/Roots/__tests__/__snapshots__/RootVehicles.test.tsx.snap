// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no items 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <RootVehicles
    isItemFavourite={false}
    item={
      Object {
        "cargo_capacity": "",
        "consumables": "",
        "cost_in_credits": "",
        "crew": "",
        "films": Array [
          "test",
        ],
        "length": "",
        "manufacturer": "",
        "max_atmosphering_speed": "",
        "model": "",
        "name": "",
        "passengers": "",
        "pilots": Array [
          "test",
        ],
        "url": "",
        "vehicle_class": "",
      }
    }
    setIsItemFavourite={[Function]}
  >
    <div>
      <styled.div
        onClick={[Function]}
      >
        <StyledComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bdVaJa",
                "isStatic": false,
                "lastClassName": "hEXTsB",
                "rules": Array [
                  "
  margin-left: 3px;
  display: flex;
  justify-content: flex-end;
  cursor: pointer;
",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-bdVaJa",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
          onClick={[Function]}
        >
          <div
            className="sc-bdVaJa hEXTsB"
            onClick={[Function]}
          >
            <ForwardRef(FavoriteBorderIcon)>
              <ForwardRef(SvgIcon)
                data-testid="FavoriteBorderIcon"
              >
                <MuiSvgIconRoot
                  aria-hidden={true}
                  as="svg"
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium"
                  data-testid="FavoriteBorderIcon"
                  focusable="false"
                  ownerState={
                    Object {
                      "children": <path
                        d="M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z"
                      />,
                      "color": "inherit",
                      "component": "svg",
                      "data-testid": "FavoriteBorderIcon",
                      "fontSize": "medium",
                      "viewBox": "0 0 24 24",
                    }
                  }
                  viewBox="0 0 24 24"
                >
                  <Noop />
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="FavoriteBorderIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z"
                    />
                  </svg>
                </MuiSvgIconRoot>
              </ForwardRef(SvgIcon)>
            </ForwardRef(FavoriteBorderIcon)>
          </div>
        </StyledComponent>
      </styled.div>
      <div>
        <p>
          Vehicle Name: 
        </p>
        <p>
          Vehicle Model: 
        </p>
        <p>
          Manufacturer: 
        </p>
        <p>
          Cargo Capacity: 
        </p>
        <p>
          Consumables: 
        </p>
        <p>
          Cost in Credits: 
        </p>
        <p>
          Crew Size: 
        </p>
        <p>
          Length: 
        </p>
        <p>
          Maximum Atmospheric Speed: 
        </p>
        <p>
          Passengers: 
        </p>
        <p>
          Vehicle Class: 
        </p>
      </div>
      <Modal
        displayName="Films"
        item={
          Array [
            "test",
          ]
        }
        result={[Function]}
      >
        <div>
          <Styled(Button)
            color="secondary"
            onClick={[Function]}
            tag="button"
            type="button"
          >
            <Noop />
            <Button
              className="css-bjdol9"
              color="secondary"
              onClick={[Function]}
              tag="button"
              type="button"
            >
              <button
                aria-label={null}
                className="css-bjdol9 btn btn-secondary"
                onClick={[Function]}
                type="button"
              >
                Films
              </button>
            </Button>
          </Styled(Button)>
          <Styled(ForwardRef(ModalUnstyled))
            BackdropComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "__emotion_base": "div",
                "__emotion_forwardProp": [Function],
                "__emotion_real": [Circular],
                "__emotion_styles": Array [
                  "
  z-index: -1;
  position: fixed;
  right: 0;
  bottom: 0;
  top: 0;
  left: 0;
",
                  [Function],
                  "",
                ],
                "defaultProps": undefined,
                "displayName": "Styled(div)",
                "render": [Function],
                "withComponent": [Function],
              }
            }
            aria-describedby="unstyled-modal-description"
            aria-labelledby="unstyled-modal-title"
            onClose={[Function]}
            open={false}
          >
            <Noop />
            <ForwardRef(ModalUnstyled)
              BackdropComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "__emotion_base": "div",
                  "__emotion_forwardProp": [Function],
                  "__emotion_real": [Circular],
                  "__emotion_styles": Array [
                    "
  z-index: -1;
  position: fixed;
  right: 0;
  bottom: 0;
  top: 0;
  left: 0;
",
                    [Function],
                    "",
                  ],
                  "defaultProps": undefined,
                  "displayName": "Styled(div)",
                  "render": [Function],
                  "withComponent": [Function],
                }
              }
              aria-describedby="unstyled-modal-description"
              aria-labelledby="unstyled-modal-title"
              className="css-497lzk"
              onClose={[Function]}
              open={false}
            />
          </Styled(ForwardRef(ModalUnstyled))>
        </div>
      </Modal>
      <Modal
        displayName="Pilots"
        item={
          Array [
            "test",
          ]
        }
        result={[Function]}
      >
        <div>
          <Styled(Button)
            color="secondary"
            onClick={[Function]}
            tag="button"
            type="button"
          >
            <Noop />
            <Button
              className="css-bjdol9"
              color="secondary"
              onClick={[Function]}
              tag="button"
              type="button"
            >
              <button
                aria-label={null}
                className="css-bjdol9 btn btn-secondary"
                onClick={[Function]}
                type="button"
              >
                Pilots
              </button>
            </Button>
          </Styled(Button)>
          <Styled(ForwardRef(ModalUnstyled))
            BackdropComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "__emotion_base": "div",
                "__emotion_forwardProp": [Function],
                "__emotion_real": [Circular],
                "__emotion_styles": Array [
                  "
  z-index: -1;
  position: fixed;
  right: 0;
  bottom: 0;
  top: 0;
  left: 0;
",
                  [Function],
                  "",
                ],
                "defaultProps": undefined,
                "displayName": "Styled(div)",
                "render": [Function],
                "withComponent": [Function],
              }
            }
            aria-describedby="unstyled-modal-description"
            aria-labelledby="unstyled-modal-title"
            onClose={[Function]}
            open={false}
          >
            <Noop />
            <ForwardRef(ModalUnstyled)
              BackdropComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "__emotion_base": "div",
                  "__emotion_forwardProp": [Function],
                  "__emotion_real": [Circular],
                  "__emotion_styles": Array [
                    "
  z-index: -1;
  position: fixed;
  right: 0;
  bottom: 0;
  top: 0;
  left: 0;
",
                    [Function],
                    "",
                  ],
                  "defaultProps": undefined,
                  "displayName": "Styled(div)",
                  "render": [Function],
                  "withComponent": [Function],
                }
              }
              aria-describedby="unstyled-modal-description"
              aria-labelledby="unstyled-modal-title"
              className="css-497lzk"
              onClose={[Function]}
              open={false}
            />
          </Styled(ForwardRef(ModalUnstyled))>
        </div>
      </Modal>
    </div>
  </RootVehicles>
</Provider>
`;
